iguana.corecontroller/src/main/java/org/aksw/iguana/cc/utils/IndexedQueryReader.java

```diff
@@ -0,0 +1,165 @@
+        return new IndexedQueryReader(filepath, FileUtils.getLineEnding(filepath));
+    }
+
+    /**
+     * Creates an object that indexes each content in between two of the given separators (including the beginning and
+     * end of the given file). <br/>
+     * @param filepath path to the file
+     * @param separator the separator for each bundle
+     * @throws IOException
+     */
+    private IndexedQueryReader(String filepath, String separator) throws IOException {
+        this.file = new File(filepath);
+        this.indexFile(separator);
+    }
+
+    /**
+     * Returns the indexed content with the given index.
+     * @param index the index of the searched content
+     * @return the searched content
+     * @throws IOException
+     */
+    public String readQuery(int index) throws IOException {
+        // split long value into two integers
+        long length = this.indices.get(index)[1];
+        int upper = (int)(length >> 32);
+        int lower = (int)length;
+
+        // store content into two byte arrays (for the case that the content is >~2GB)
+        byte[] data1 = new byte[lower];
+        byte[] data2 = new byte[upper];
+        String output;
+        try(RandomAccessFile raf = new RandomAccessFile(this.file, "r")) {
+            raf.seek(this.indices.get(index)[0]);
+            raf.read(data1);
+            output = new String(data1, StandardCharsets.UTF_8);
+            raf.read(data2);
+            output += new String(data2, StandardCharsets.UTF_8);
+        }
+        return output;
+    }
+
+    /**
+     * This method returns a list of strings that contains every indexed content.
+     * @return list of lines
+     * @throws IOException
+     */
+    public List<String> readQueries() throws IOException {
+        ArrayList<String> out = new ArrayList<>();
+        for(int i = 0; i < indices.size(); i++) {
+            out.add(this.readQuery(i));
+        }
+        return out;
+    }
+
+    /**
+     * Returns the number of indexed content.
+     * @return number of indexed objects
+     */
+    public int size() {
+        return this.indices.size();
+    }
+
+    /**
+     * Indexes every content in between two of the given separator. The beginning and the end of the file count as
+     * separators too.
+     * @param separator the custom separator
+     * @throws IOException
+     */
+    private void indexFile(String separator) throws IOException {
+        this.indices = new ArrayList<>();
+        try(FileReader fr = new FileReader(this.file, StandardCharsets.UTF_8);
+            BufferedReader br = new BufferedReader(fr)) {
+            // The method needs to know the length of the line ending used in the file to be able to properly calculate
+            // the starting byte position of a line
+            long lastIndex = 0;
+            long currentIndex = 0;
+            int counter = 0;
+            int character;
+            boolean isWhiteSpace = true;
```

Comment:
I know it was something I said that we should remove leading and trailing white spaces. It seems to make it unnecessary complicated. Did we would want them removed?